src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -0,0 +1,425 @@
+        /// </summary>
+        /// <remarks>
+        /// <para>Some syntax nodes, such as parenthesized expressions, require a matching closing delimiter to end the
+        /// syntax node. If this node is omitted from the source code, the parser will automatically insert a zero-width
+        /// "missing" closing delimiter token to produce a valid syntax tree. This method determines if required closing
+        /// delimiters are present in the original source.</para>
+        /// </remarks>
+        /// <param name="currentNode"></param>
+        /// <param name="lastDelimiterSpan"></param>
+        /// <returns>
+        /// <list type="bullet">
+        /// <item><description><see langword="true"/> if <paramref name="currentNode"/> requires a closing delimiter and the closing delimiter is present in the source (i.e. not missing)</description></item>
+        /// <item><description><see langword="true"/> if <paramref name="currentNode"/> does not require a closing delimiter</description></item>
+        /// <item><description>otherwise, <see langword="false"/>.</description></item>
+        /// </list>
+        /// </returns>
+        private static bool ClosingDelimiterExistsIfNeeded(SyntaxNode currentNode, ref TextSpan lastDelimiterSpan)
+        {
+            switch (currentNode.Kind())
+            {
+                case SyntaxKind.ArgumentList:
+                    var argumentList = (ArgumentListSyntax)currentNode;
+                    lastDelimiterSpan = argumentList.CloseParenToken.Span;
+                    return !argumentList.CloseParenToken.IsMissing;
+
+                case SyntaxKind.ParenthesizedExpression:
+                    var parenthesizedExpression = (ParenthesizedExpressionSyntax)currentNode;
+                    lastDelimiterSpan = parenthesizedExpression.CloseParenToken.Span;
+                    return !parenthesizedExpression.CloseParenToken.IsMissing;
+
+                case SyntaxKind.BracketedArgumentList:
+                    var bracketedArgumentList = (BracketedArgumentListSyntax)currentNode;
+                    lastDelimiterSpan = bracketedArgumentList.CloseBracketToken.Span;
+                    return !bracketedArgumentList.CloseBracketToken.IsMissing;
+
+                case SyntaxKind.ObjectInitializerExpression:
+                    var initializerExpressionSyntax = (InitializerExpressionSyntax)currentNode;
+                    lastDelimiterSpan = initializerExpressionSyntax.CloseBraceToken.Span;
+                    return !initializerExpressionSyntax.CloseBraceToken.IsMissing;
+
+                default:
+                    // Type of node does not require a closing delimiter
+                    return true;
+            }
+        }
+
+        private static bool ReachedSemicolonSyntax(SyntaxNode currentNode, ISyntaxFactsService syntaxFacts, ref bool finalDelimiterNeedsSemicolon)
+        {
+            finalDelimiterNeedsSemicolon = false;
```

Comment:
this  variable is just weird. at every step fo the loop that calls this we set it back to false, but only set it to true if we finally get a 'do' statement.  but if we get a do-statement we can break out of the loop.  So, we could remove this variableentirely and just have the caller just say `if (currentNode.Kind() == SyntaxKind.DoStatement)`

Reply:
➡️ Seems to no longer apply